Common/Utils/include/CommonUtils/StringUtils.h

```diff
@@ -89,6 +92,42 @@ std::string concat_string(Ts const&... ts)
   return s.str();
 }
 
+// Check if the path exists
+static inline bool pathExists(const std::string_view p)
+{
+  struct stat buffer;
+  return (stat(p.data(), &buffer) == 0);
+}
+
+// Check if the path is a directory
+static inline bool pathIsDirectory(const std::string_view p)
+{
+  struct stat buffer;
+  return (stat(p.data(), &buffer) == 0) && S_ISDIR(buffer.st_mode);
+}
+
+static inline std::string getFullPath(const std::string_view p)
+{
+  std::unique_ptr<char[]> real_path(realpath(p.data(), nullptr));
+  return std::string(real_path.get());
+}
```

Comment:
we also use boost::filesystem in various places. I have the impression, that some of the proposed functionality in the PR is directly available there.

Reply:
@sawenzel the only functionality this PR adds is the possibility to alter IO directories in  ConfigurableParam
 (will add also direct setters). The lines you commented are simply moved from the NameConf to StringUtils to avoid dependencies problem (including your method pathExists). Later I will convert the StringUtils to static class to avoid inlining heavy methods.